locksmith/src/fulfillment/dispatcher.ts

```diff
@@ -16,10 +16,14 @@ interface KeyToGrant {
   expiration?: number
 }
 export default class Dispatcher {
-  async getPurchaser(network: number) {
-    const provider = new ethers.providers.JsonRpcProvider(
+  providerForNetwork(network: number) {
+    const provider = new ethers.providers.JsonRpcBatchProvider(
```

Comment:
The dispatcher sends transactions, so I am not sure batching makes much sense here!

Reply:
Yeah, I replaced the use here for consistency since it would affect the behavior either way.

Reply:
What do you think of _not_ changing it for now then so we can test more thoroughly and avoid breaking things in unexpected ways.